cirq/google/merge_rotations.py

```diff
@@ -65,29 +62,28 @@ def _scan_single_qubit_ops(
             self,
             circuit: Circuit,
             index: Optional[int],
-            qubit: ops.QubitId) -> Tuple[List[int], List[ops.KnownMatrix]]:
-        operations = []  # type: List[ops.KnownMatrix]
+            qubit: ops.QubitId) -> Tuple[List[int], List[ops.Operation]]:
+        operations = []  # type: List[ops.Operation]
         indices = []  # type: List[int]
         while index is not None:
             op = cast(ops.Operation, circuit.operation_at(qubit, index))
             if len(op.qubits) != 1:
                 break
-            operation = self.extensions.try_cast(ops.KnownMatrix, op)
-            if operation is None:
+            if protocols.unitary(op, None) is None:
                 break
             indices.append(index)
-            operations.append(operation)
+            operations.append(op)
```

Comment:
Might want to capture the unitary matrices here and return them as well. Then they don't have to be recomputed inside `_merge_rotations`. Could also compute the matrix product as we go to avoid storing all the matrices separately.

Reply:
I would, but see the is-it-one-xmon-op check? Need the operation to do that check.

Reply:
Yes, would have to return the list of ops _and_ the matrices (or the final matrix if you compute the matrix products as you go through the loop here).